### PR TITLE
[SDTEST-116] Implement manual_api_events metric

### PR DESCRIPTION
**Pull request by anmarchenko**
**What does this PR do?**
Tracks manual_api_events metric when public API is used.

Internal instrumentations don't use public API anymore.

**How to test the change?**
Unit tests are provided

### CODE REVIEW

File: lib/datadog/ci/contrib/cucumber/formatter.rb [hunk 1]
```diff
@@ -70,7 +70,7 @@ def on_test_case_started(event)
 
             start_test_suite(test_suite_name) unless same_test_suite_as_current?(test_suite_name)
 
-            test_span = CI.start_test(
```
Comment (nikita-tkachenko-datadog) [inline lib/datadog/ci/contrib/cucumber/formatter.rb:73]: I'm not sure what the difference is between `start_test` and `trace_test` (other than the `&block` parameter, which does not seem to be used here), is this change intentional?

**Review by nikita-tkachenko-datadog — APPROVED** [review]

File: lib/datadog/ci/contrib/cucumber/formatter.rb (hunk 1 above)
Comment (anmarchenko) [reply lib/datadog/ci/contrib/cucumber/formatter.rb:73]: Yes, the `start_test` is a convenience method that does not take a code block. I implemented it for public API, but not for private API. Now because of telemetry I cannot use public API anymore, so no convenience for me :(
